x-pack/plugins/metrics_entities/server/services/install_mappings.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ElasticsearchClient } from 'kibana/server';
+
+import { Mappings } from '../modules/types';
+import type { Logger } from '../../../../../src/core/server';
+
+import { computeMappingId } from './utils';
+import { getIndexExists } from './utils/get_index_exists';
+import { logMappingDebug } from './utils/log_mapping_debug';
+import { logMappingError } from './utils/log_mapping_error';
+
+interface CreateMappingOptions {
+  esClient: ElasticsearchClient;
+  mappings: Mappings[];
+  prefix: string;
+  suffix: string;
+  logger: Logger;
+}
+
+export const installMappings = async ({
+  esClient,
+  mappings,
+  prefix,
+  suffix,
+  logger,
+}: CreateMappingOptions): Promise<void> => {
+  for (const mapping of mappings) {
+    const { index } = mapping.mappings._meta;
+    const mappingId = computeMappingId({ id: index, prefix, suffix });
+    const exists = await getIndexExists(esClient, mappingId);
+    const computedBody = {
+      ...mapping,
+      ...{
+        mappings: {
+          ...mapping.mappings,
+          _meta: {
+            ...mapping.mappings._meta,
+            // TODO: Do we want to keep the version as 1.0.0 here? If so then this should be moved into constants
+            ...{ created_by: 'metrics_entities', index: mappingId, version: { created: '1.0.0' } },
+          },
+        },
+      },
+    };
+    if (!exists) {
+      try {
+        logMappingDebug({ id: mappingId, logger, message: 'does not exist, creating the mapping' });
```

Comment:
May be worthy of an `info` logger here

Reply:
Yeah, when it doesn't exist, this should be info rather than debug. That's a good call. I will change it here in this PR now.